### ISSUE TITLE
Ignoring broken variables/instructions on recompile

### ISSUE THREAD

**Issue by KtntKot** [Col-E/Recaf#374]
## Ignoring broken variables on recompile
![](https://niggaa.cat/raw/API2WWQ3qxvCqj.png?__cf_chl_jschl_tk__=5425eefbad6434df5c5a988ac17204079609796a-1619180720-0-Aeg400_T__3nCYtLB498GgYklrF_GGDfd-fhkX1j1KThUAKkl1DaCgCF1SxkNXazdnRQ0tWkyRQjgntziA1JO3m5p4qACOHR5rRwDys-S-cbU38ga5D1WJIVF5W1QBotxQgVT0qE3aDUcmnomU2HdSV0PBdfOHYbvT5N0RbU9syyG0kAbA8O6blFsWTqg76L_icVjQnc30NlbtfexSnd5vyRpsaJsJa7iD505rxtNSn0XgEmCIouMW-XMVa6wTJkYkemCD-tPRb42wovRtVrv050dTGNXQtS-cq6dsErQYF6FeAcKw78d3RstQm1WXw4BOexx1oCP7jk2tPG8KXpqYjBl1WXz0k6UJqhrm0-U25hj32oWNp7CbQ5fY2EWeZLqGW0xTOeFw1QdnlLvT6RGKKmTnPPsMxz8pF-CdmMOxUCf2rqpRYngY0JahhtGjzSJeuTjXc30lt8txREnbAVeNw)

I don't know if there is a way to ignore these, i didn't find one.
These instructions/variables break the recompiling, i am looking for a method of ignoring it/clearing.

**Comment by KtntKot** [comment]
‌https://niggaa.cat/API2WWQ3qxvCqj.png

**Comment by codingsocks** [comment]
Obfuscated code generally can't be recompiled. There's too many edge cases and broken stuff that generally recompiling just has too many issues to do it. You'll see this issue in other tools that offer recompilation. Since Recaf recompiles the entire class on save, any concepts that can only occur in a bytecode form can't exactly be recompiled since they produce invalid Java. This is exactly why the assembler feature in Recaf exists. You can read the documentation on it here: https://www.coley.software/Recaf-documentation/use-assembler.html

I'd highly recommend you check it out and try to not rely on the recompilation feature, since it won't always be there for you.

**Comment by KtntKot** [comment]
yes you can't really decompile it but there are some cases, even a lot of them.

**Comment by KtntKot** [comment]
in this case i made simple asm cleaner to clean out these instructions and then manually removed rest of them.
Maybe advanced bytecode manipulation like in jbytemod?
regards.

**Comment by KtntKot** [comment]
edit: nvm i am blind thanks for help have a nice day
